From f17223459b3cea19ebafe0c22aa580c3b382642e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:23 -0400
Subject: [ethernet] i40e: add vsi x-cast stats

Message-id: <1409658042-22689-143-git-send-email-sassmann@redhat.com>
Patchwork-id: 90336
O-Subject: [RHEL7.1 PATCH 142/281] i40e: add vsi x-cast stats
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Add VSI HW stats for unicast, multicast, and broadcast for Rx and Tx.
Stop printing the netdev multicast value because it doesn't include Tx
and would be confusing.

Change-ID: I08278b6657e7c838fd29a4a1f305f78fe1b150be
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 418631d4c5c2a7d39cf43cb0d07de639835777bd)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index 2744f9f..509f92c 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -58,7 +58,6 @@ static const struct i40e_stats i40e_gstrings_net_stats[] = {
  I40E_NETDEV_STAT(tx_errors),
  I40E_NETDEV_STAT(rx_dropped),
  I40E_NETDEV_STAT(tx_dropped),
- I40E_NETDEV_STAT(multicast),
  I40E_NETDEV_STAT(collisions),
  I40E_NETDEV_STAT(rx_length_errors),
  I40E_NETDEV_STAT(rx_crc_errors),
@@ -80,9 +79,13 @@ static const struct i40e_stats i40e_gstrings_veb_stats[] = {
 };
 
 static const struct i40e_stats i40e_gstrings_misc_stats[] = {
- I40E_VSI_STAT("rx_unknown_protocol", eth_stats.rx_unknown_protocol),
+ I40E_VSI_STAT("rx_unicast", eth_stats.rx_unicast),
+ I40E_VSI_STAT("tx_unicast", eth_stats.tx_unicast),
+ I40E_VSI_STAT("rx_multicast", eth_stats.rx_multicast),
+ I40E_VSI_STAT("tx_multicast", eth_stats.tx_multicast),
  I40E_VSI_STAT("rx_broadcast", eth_stats.rx_broadcast),
  I40E_VSI_STAT("tx_broadcast", eth_stats.tx_broadcast),
+ I40E_VSI_STAT("rx_unknown_protocol", eth_stats.rx_unknown_protocol),
 };
 
 static int i40e_add_fdir_ethtool(struct i40e_vsi *vsi,
-- 
1.7.1